webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/design/SiteDesignActionHandler.java

```diff
@@ -0,0 +1,660 @@
+  private String parseCodeSupprHref(WebSiteSessionController scc, String code,
+      SettingBundle settings, String currentPath) {
+    String avant;
+    String apres;
+    int index;
+    String href = "<A href=\""; /* longueur de chaine = 9 */
+
+    String finChemin;
+    String fichier;
+    String deb;
+    String theReturn = "";
+
+    index = code.indexOf(href);
+    if (index == -1) {
+      theReturn = code;
+    } else {
+
+      avant = code.substring(0, index + 9);
+
+      apres = code.substring(index + 9);
+
+      if (apres.substring(0, 7).equals("http://")) { /* lien externe */
+
+        theReturn = avant + parseCodeSupprHref(scc, apres, settings, currentPath);
+      } else if (apres.substring(0, 6).equals("ftp://")) { /* lien externe */
+
+        theReturn = avant + parseCodeSupprHref(scc, apres, settings, currentPath);
+      } else if (apres.substring(0, 3).equals("rr:")) { /* deja en relatif */
+
+        apres = apres.substring(3);
+
+        theReturn = avant + parseCodeSupprHref(scc, apres, settings, currentPath);
+      } else if (apres.substring(0, 3).equals("aa:")) {
+        // lien absolu a transformer en relatif
+
+        /* finChemin = rep/coucou.html">... */
+        finChemin = code.substring(index + 9 + 3);
+
+
+        /* traitement */
+        int indexGuillemet = finChemin.indexOf("\"");
+
+
+        /* absolute = rep/coucou.html */
+        String absolute = finChemin.substring(0, indexGuillemet);
+
+
+        /* apres = ">... */
+        apres = finChemin.substring(indexGuillemet);
+
+        int indexSlash = absolute.lastIndexOf("\\");
+
+        if (indexSlash == -1) {
+          // pas d'arborescence, le fichier du lien est sur la racine
+          fichier = absolute;
+          deb = "";
+        } else {
+          /* fichier = coucou.html */
+          fichier = absolute.substring(indexSlash + 1);
+          deb = absolute.substring(0, indexSlash);
+        }
+        List<String> tab = buildTab(deb + "/");
+        // dans ce tableau il manque l'id
+
+        /* cheminContexte = id/rep */
+        int longueur = scc.getComponentId().length();
+        int index2 = currentPath.lastIndexOf(scc.getComponentId());
+        String chemin = currentPath.substring(index2 + longueur);
+
+        chemin = chemin.substring(1);
+        chemin = removeDoubleAntiSlash(chemin);
+        String cheminContexte = chemin;
+        List<String> tabContexte = buildTab(cheminContexte + "/");
+        /* ajoute l'id dans le premier tableau */
+        tab.add(0, tabContexte.get(0));
+
+        /* tabCommun = [id | rep] */
+        List<String> tabCommun = sortCommun(tabContexte, tab);
+
+        /* reste = vide */
+        String reste = sortReste(tab, tabCommun);
+
+        /* nbPas = 0 */
+        int nbPas = tabContexte.size() - tabCommun.size();
+        String relatif = "";
+        int i = 0;
+        while (i < nbPas) {
+          relatif += "../";
+          i++;
+        }
+
+        if (reste.equals("")) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)